kasa/device_factory.py

```diff
@@ -129,3 +133,37 @@ def get_device_class_from_info(info: Dict[str, Any]) -> Type[SmartDevice]:
 
         return SmartBulb
     raise UnsupportedDeviceException("Unknown device type: %s" % type_)
+
+
+def get_device_class_from_type_name(device_type: str) -> Optional[Type[SmartDevice]]:
+    """Return the device class from the type name."""
+    supported_device_types: dict[str, Type[SmartDevice]] = {
+        "SMART.TAPOPLUG": TapoPlug,
+        "SMART.KASAPLUG": TapoPlug,
+        "IOT.SMARTPLUGSWITCH": SmartPlug,
+    }
+    return supported_device_types.get(device_type)
+
+
+def get_protocol_from_connection_name(
+    connection_name: str, host: str, credentials: Optional[Credentials] = None
+) -> Optional[TPLinkProtocol]:
+    """Return the protocol from the connection name."""
+    supported_device_protocols: dict[
+        str, Tuple[Type[TPLinkProtocol], Type[TPLinkTransport]]
+    ] = {
+        "IOT.KLAP": (TPLinkIotProtocol, TPLinkKlapTransport),
+        "SMART.AES": (TPLinkSmartProtocol, TPLinkAesTransport),
+        "SMART.KLAP": (TPLinkSmartProtocol, TPlinkKlapTransportV2),
```

Comment:
Tbf, I don't like the usage of a "magic" string to proto & transport mapping, but that's something we may need to figure out in another PR to keep this tidier?

Reply:
Ok agreed.  Let's add the `ConnectionParameters` in the next PR